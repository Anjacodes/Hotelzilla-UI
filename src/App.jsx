import React, { useEffect } from 'react';
import { Route, Routes } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import SignUp from './components/SignUp';
import Home from './components/Home';
import ProtectedRoute from './components/Details/ProtectedRoute';
import Index from './components/Index';
import Reserve from './components/Reserve';
import Login from './components/Login';
import { login } from './redux/login/login';
import {
  isLoggedIn,
  isLoggedOut,
  getRole,
  getUserId,
} from './modules/auth-module';

const tokenKey = 'token';

const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    if (localStorage.getItem(tokenKey)) {
      dispatch(login(JSON.parse(localStorage.getItem(tokenKey))));
    }
  }, []);

  const token = useSelector((state) => state.login.token);

  useEffect(() => {
    console.log(token);
    console.log(isLoggedIn(token));
    console.log(isLoggedOut(token));
    console.log(getRole(token));
    console.log(getUserId(token));
  }, [token]);

  const state = {
    loggedIn: true,
  };

  return (
    <>
      <Routes>
        <Route path="/register" element={<SignUp />} />
        <Route path="/login" element={<Login />} />
        <Route path="/" element={<Home />}>
          <Route index element={<Index />} />
          <Route path="reserve" element={<Reserve />} />
          <Route
            path=":roomId"
            element={<ProtectedRoute loggedIn={state.loggedIn} />}
          />
        </Route>
      </Routes>
    </>
  );
};

export default App;
