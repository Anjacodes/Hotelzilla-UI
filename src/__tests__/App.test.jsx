/**
 * @vitest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { it, describe, expect } from 'vitest';
import store from '../redux/configStore';
import App from '../App';

describe('App', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
