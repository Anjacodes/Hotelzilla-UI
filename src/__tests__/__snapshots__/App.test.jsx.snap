// Vitest Snapshot v1

exports[`App > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex w-screen overflow-y-hidden"
      >
        <button
          class="absolute top-2 left-2 z-20 rounded-md border-2 p-1 md:hidden"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium h-4 w-4 css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </button>
        <nav
          class="hidden absolute top-0 left-0 z-10 h-screen w-3/5 flex-col items-center justify-evenly overflow-hidden border-r bg-white py-4 pl-2 md:static md:flex md:w-1/5"
        >
          <a
            class="px-4 md:px-6 lg:px-8"
            href="/"
          >
            <img
              alt="Hotelzilla Logo"
              src="Hotelzilla-logo.png"
            />
          </a>
          <div
            class="flex w-full flex-col pl-2 pt-12 text-gray-600"
          >
            <a
              class="mt-8 py-2 font-Taxicab text-xl font-bold hover:bg-lime-400 hover:text-slate-50"
              href="/login"
            >
              LOG IN
            </a>
            <a
              class="py-2 font-Taxicab text-xl font-bold hover:bg-lime-400 hover:text-slate-50"
              href="/register"
            >
              SIGN UP
            </a>
          </div>
          <footer
            class="mt-auto flex flex-col items-center"
          >
            <div
              class="mb-2 flex w-full justify-around"
            >
              <i
                class="fa-brands fa-twitter text-gray-600"
              />
              <i
                class="fa-brands fa-facebook-f text-gray-600"
              />
              <i
                class="fa-brands fa-google-plus-g text-gray-600"
              />
              <i
                class="fa-brands fa-vimeo-v text-gray-600"
              />
              <i
                class="fa-brands fa-pinterest-p text-gray-600"
              />
            </div>
            <p
              class="text-xs text-gray-600"
            >
              ©2022 Hotelzilla
            </p>
          </footer>
        </nav>
        <div
          class="w-full bg-slate-100 md:w-4/5"
        >
          <div
            class="flex items-center"
          >
            <div
              class="flex h-screen items-center justify-center"
            >
              <i
                class="fa-solid fa-caret-left absolute left-0 h-[50px] rounded-r-full bg-gray-200 py-4 px-6 text-slate-50 md:left-[20%] hover:not"
              />
              <div
                class="grid w-10/12 grid-cols-1 gap-3 p-5 md:grid-cols-3 md:grid-rows-1 sm:"
              />
              <i
                class="fa-solid fa-caret-right absolute right-0 h-[50px] rounded-l-full bg-lime-400 py-4 px-6 text-slate-50 hover:not"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex w-screen overflow-y-hidden"
    >
      <button
        class="absolute top-2 left-2 z-20 rounded-md border-2 p-1 md:hidden"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium h-4 w-4 css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <nav
        class="hidden absolute top-0 left-0 z-10 h-screen w-3/5 flex-col items-center justify-evenly overflow-hidden border-r bg-white py-4 pl-2 md:static md:flex md:w-1/5"
      >
        <a
          class="px-4 md:px-6 lg:px-8"
          href="/"
        >
          <img
            alt="Hotelzilla Logo"
            src="Hotelzilla-logo.png"
          />
        </a>
        <div
          class="flex w-full flex-col pl-2 pt-12 text-gray-600"
        >
          <a
            class="mt-8 py-2 font-Taxicab text-xl font-bold hover:bg-lime-400 hover:text-slate-50"
            href="/login"
          >
            LOG IN
          </a>
          <a
            class="py-2 font-Taxicab text-xl font-bold hover:bg-lime-400 hover:text-slate-50"
            href="/register"
          >
            SIGN UP
          </a>
        </div>
        <footer
          class="mt-auto flex flex-col items-center"
        >
          <div
            class="mb-2 flex w-full justify-around"
          >
            <i
              class="fa-brands fa-twitter text-gray-600"
            />
            <i
              class="fa-brands fa-facebook-f text-gray-600"
            />
            <i
              class="fa-brands fa-google-plus-g text-gray-600"
            />
            <i
              class="fa-brands fa-vimeo-v text-gray-600"
            />
            <i
              class="fa-brands fa-pinterest-p text-gray-600"
            />
          </div>
          <p
            class="text-xs text-gray-600"
          >
            ©2022 Hotelzilla
          </p>
        </footer>
      </nav>
      <div
        class="w-full bg-slate-100 md:w-4/5"
      >
        <div
          class="flex items-center"
        >
          <div
            class="flex h-screen items-center justify-center"
          >
            <i
              class="fa-solid fa-caret-left absolute left-0 h-[50px] rounded-r-full bg-gray-200 py-4 px-6 text-slate-50 md:left-[20%] hover:not"
            />
            <div
              class="grid w-10/12 grid-cols-1 gap-3 p-5 md:grid-cols-3 md:grid-rows-1 sm:"
            />
            <i
              class="fa-solid fa-caret-right absolute right-0 h-[50px] rounded-l-full bg-lime-400 py-4 px-6 text-slate-50 hover:not"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
